src/components/products/Products.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from 'react'
+
+import ProductCard from './ProductCard/ProductCard'
+import itemsServices from '../../services/items'
+import './products.css'
+
+const Products = ({deals}) => {
+    const [items, setItems] = useState([])
+
+    useEffect(() => {
+        const fetch_data = async () => {
+            const response_data = await itemsServices.getAllItems()
+            setItems(response_data.items)
+        }
+        fetch_data()        
+    }, [])
+
+    if (deals) {
+
+        const computed_deals_items = items.filter(item => item.isOnSale)
```

Comment:
It's best to let the API filter items for performance, and to make sure you are actually getting all of the sale items. In most cases, you will probably want to load the data in the page component and pass it down to the ProductCard component instead of loading the data in the product card component.

Reply:
Yes that's so true